src/StreamJsonRpc/CorrelationManagerTracingStrategy.cs

```diff
@@ -0,0 +1,131 @@
+
+                if (this.TraceSource is object && (this.TraceSource.Switch.Level & SourceLevels.ActivityTracing) == SourceLevels.ActivityTracing && this.TraceSource.Listeners.Count > 0)
+                {
+                    traceparent.Flags |= TraceParent.TraceFlags.Sampled;
+                }
+
+                request.TraceParent = traceparent.ToString();
+                request.TraceState = TraceState;
+            }
+        }
+
+        /// <inheritdoc/>
+        public unsafe IDisposable? ApplyInboundActivity(JsonRpcRequest request)
+        {
+            Requires.NotNull(request, nameof(request));
+
+            if (request.TraceParent is null)
+            {
+                return null;
+            }
+
+            var traceparent = new TraceParent(request.TraceParent);
+            Guid childActivityId = Guid.NewGuid();
+            string? activityName = this.GetInboundActivityName(request);
+
+            return new ActivityState(request, this.TraceSource, activityName, traceparent.TraceIdGuid, childActivityId);
+        }
+
+        /// <summary>
+        /// Determines the name to give to the activity started when dispatching an incoming RPC call.
+        /// </summary>
+        /// <param name="request">The inbound RPC request.</param>
+        /// <returns>The name of the activity.</returns>
+        /// <remarks>
+        /// The default implementation uses <see cref="JsonRpcRequest.Method"/> as the activity name.
+        /// </remarks>
+        protected virtual string? GetInboundActivityName(JsonRpcRequest request) => request?.Method;
+
+        private static unsafe void FillRandomBytes(Span<byte> buffer) => CopyGuidToBuffer(Guid.NewGuid(), buffer);
+
+        private static unsafe void CopyGuidToBuffer(Guid guid, Span<byte> buffer)
+        {
+            Assumes.True(buffer.Length <= sizeof(Guid));
+            ReadOnlySpan<byte> guidBytes = new ReadOnlySpan<byte>(&guid, sizeof(Guid));
+            guidBytes.Slice(0, buffer.Length).CopyTo(buffer);
+        }
+
+        private class ActivityState : IDisposable
+        {
+            private readonly TraceSource? traceSource;
+            private readonly Guid originalActivityId;
+            private readonly string? originalTraceState;
+            private readonly string? activityName;
+            private readonly Guid parentTraceId;
+
+            internal ActivityState(JsonRpcRequest request, TraceSource? traceSource, string? activityName, Guid parentTraceId, Guid childTraceId)
+            {
+                this.originalActivityId = Trace.CorrelationManager.ActivityId;
+                this.originalTraceState = TraceState;
+                this.activityName = activityName;
+                this.parentTraceId = parentTraceId;
+
+                if (traceSource is object)
+                {
+                    Trace.CorrelationManager.ActivityId = parentTraceId;
```

Comment:
Typically, yes. But not this time, because we immediately set `ActivityId` to something *else* on the very next line. So the only reason we set it here is *for* the `Transfer` call.
I'll add a code comment to indicate this.